### PR TITLE
Handle Events

### PR DESCRIPTION
**Pull request by migo1**
### MILESTONES

- Handle events in a React app.
- Use React life cycle methods.
- Use states and props.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,14 +1,23 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Button from './Button';
 import './calculator.css';
+import caclulate from '../logic/calculate';
 
 const btnKeys = ['AC', '+/-', '%', '÷', '7', '8', '9', '×', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
```
Comment (AmaduKamara) [inline src/components/Calculator.js]: - [ ] I noticed you must have used the wrong multiplication sign which is causing the multiplication operation to fail. When I changed it and used the X sign, it worked. Maybe you copied the sign and pasted it or so. Kindly remove the current 'x' sign and type in X manually from your keyboard and test your app. Please fix that 👍🏽

Screenshot 2023-07-06 at 7 40 44 AM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @migo1,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ You used the logic files provided to make the calculator work 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] It would be nice if you could work on showing all the numbers clicked for a math operation and then clicking the "=" sign to get the result. Currently, only the first number is shown (for instance 9 + 3 - on 9 shows on the screen, and when 3 is clicked, it does not show on the screen) on the screen. Please work on that so one sees both numbers when clicked for a math operation before hitting the "=" sign to get a result. 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by migo1** [comment]
@AmaduKamara noted

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
 
 function Calculator() {
+  const [state, setState] = useState({ total: null, next: null, operation: null });
+
+  const { total, next, operation } = state;
+
+  const handleClick = (e) => {
+    const btnValue = e.target.innerHTML;
+    setState(caclulate(state, btnValue));
+  };
```
Comment (OyePriscilla) [inline src/components/Calculator.js:16]: - [ ] As stated in the review summary template earlier, the second operand is not yet being displayed on the screen. For example if I calculate `2 + 5`, only figure 2 was displayed to the screen; kindly work on it. 👍

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @migo1 👍, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Calulator operator `X` worked as expected 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 2 above)
Comment (migo1) [reply src/components/Calculator.js:16]: noted, thanks for that review

**Review by binhussen — APPROVED** [review]
Hi @migo1 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Operate Calculator Events ✔️
- Well Structured code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**